es5-persistence/src/main/java/com/netflix/conductor/dao/es5/index/ElasticSearchDAOV5.java

```diff
@@ -627,4 +630,73 @@ private void addSortOptionToSearchRequest(SearchRequestBuilder searchRequestBuil
             .collect(Collectors.toCollection(LinkedList::new));
     }
 
+    @Override
+    public List<Message> getMessages(String queue) {
+        try {
+            Expression expression = Expression.fromString("queue='" + queue + "'");
+            QueryBuilder queryBuilder = expression.getFilterBuilder();
+
+            BoolQueryBuilder filterQuery = QueryBuilders.boolQuery().must(queryBuilder);
+            QueryStringQueryBuilder stringQuery = QueryBuilders.queryStringQuery("*");
+            BoolQueryBuilder fq = QueryBuilders.boolQuery().must(stringQuery).must(filterQuery);
+
+            final SearchRequestBuilder srb = elasticSearchClient.prepareSearch(logIndexPrefix + "*")
+                    .setQuery(fq)
+                    .setTypes(MSG_DOC_TYPE)
+                    .addSort(SortBuilders.fieldSort("created").order(SortOrder.ASC));
+
+            return mapGetMessagesResponse(srb.execute().actionGet());
+        } catch (Exception e) {
+            logger.error("Failed to get messages for queue: {}", queue, e);
+        }
+        return null;
```

Comment:
Please throw an exception instead of returning null.

Reply:
Done.